python/tank/platform/environment.py

```diff
@@ -290,9 +289,7 @@ def get_framework_descriptor(self, framework_name):
                             "key for engine %s" % (self._env_path, framework_name))
 
         # get the descriptor object for the location
-        d = descriptor.get_from_location(descriptor.AppDescriptor.FRAMEWORK,
-                                         self.__pipeline_config,
-                                         location_dict)
+        d = self.__pipeline_config.get_framework_descriptor(location_dict)
```

Comment:
minor: For these methods, maybe instead of `d = self.` just `return self.`? Would avoid use of nondescript `d` variable.